src/components/overview/productInfo.js

```diff
@@ -1,24 +1,45 @@
 import React from "react";
 import { useProduct } from "../../contexts/ProductContext.js";
-import FacebookIcon from '@mui/icons-material/Facebook';
-import TwitterIcon from '@mui/icons-material/Twitter';
-import PinterestIcon from '@mui/icons-material/Pinterest';
+import Rating from "@mui/material/Rating";
+import {
+  FacebookShareButton,
+  PinterestShareButton,
+  TwitterShareButton,
+} from "react-share";
+import { FacebookIcon, TwitterIcon, PinterestIcon } from "react-share";
 
 export default function ProductInfo() {
-  const { product, styleReducer } = useProduct();
+  const { product, styleReducer, meta } = useProduct();
   const [styleValue, setStyleValue] = styleReducer;
+  console.log(product);
   return (
     <div className="product-info">
       <div>
-        <p>Reviews: </p>
+        <Rating
```

Comment:
this is getting tricky to read and looks slightly unreliable. want to open a dialogue with you– how can we make this code block more readable and reliable? what are some suggestions from this PR review that can be applied here? what are some other things that are possibly unsafe?